lxd/seccomp.go

```diff
@@ -19,16 +22,93 @@ init_module errno 1
 finit_module errno 1
 delete_module errno 1
 `
-
+const COMPAT_BLOCKING_POLICY = `
+[x86_64]
+compat_sys_rt_sigaction
+stub_x32_rt_sigreturn
+compat_sys_ioctl
+compat_sys_readv
+compat_sys_writev
+compat_sys_recvfrom
+compat_sys_sendmsg
+compat_sys_recvmsg
+stub_x32_execve
+compat_sys_ptrace
+compat_sys_rt_sigpending
+compat_sys_rt_sigtimedwait
+compat_sys_rt_sigqueueinfo
+compat_sys_sigaltstack
+compat_sys_timer_create
+compat_sys_mq_notify
+compat_sys_kexec_load
+compat_sys_waitid
+compat_sys_set_robust_list
+compat_sys_get_robust_list
+compat_sys_vmsplice
+compat_sys_move_pages
+compat_sys_preadv64
+compat_sys_pwritev64
+compat_sys_rt_tgsigqueueinfo
+compat_sys_recvmmsg
+compat_sys_sendmmsg
+compat_sys_process_vm_readv
+compat_sys_process_vm_writev
+compat_sys_setsockopt
+compat_sys_getsockopt
+compat_sys_io_setup
+compat_sys_io_submit
+stub_x32_execveat
+`
 var seccompPath = shared.VarPath("security", "seccomp")
 
 func SeccompProfilePath(c container) string {
 	return path.Join(seccompPath, c.Name())
 }
 
+func ContainerNeedsSeccomp(c container) bool {
+	config := c.ExpandedConfig()
+
+	/* these are enabled by default, so if the keys aren't present, that
+	 * means "true"
+	 */
+	default_, ok := config["security.syscalls.default"]
+	if !ok || shared.IsTrue(default_) {
+		return true
+	}
+
+	compat, ok := config["security.syscalls.compat"]
+	if !ok || shared.IsTrue(compat) {
+		return true
+	}
+
+	raw := config["raw.seccomp"]
+	if raw != "" {
+		return true
+	}
+
+	return false
+}
+
 func getSeccompProfileContent(c container) string {
-	/* for now there are no seccomp knobs. */
-	return DEFAULT_SECCOMP_POLICY
+	config := c.ExpandedConfig()
+	policy := SECCOMP_HEADER
+
+	default_, ok := config["security.syscalls.default"]
```

Comment:
On Tue, May 24, 2016 at 11:39:44AM -0700, ctrlrsf wrote:
> >  func getSeccompProfileContent(c container) string {
> > -	/* for now there are no seccomp knobs. */
> > -	return DEFAULT_SECCOMP_POLICY
> > +	config := c.ExpandedConfig()
> > +	policy := SECCOMP_HEADER
> > +
> > +	default_, ok := config["security.syscalls.default"]
> 
> Is the extra underscore after default intentional here?

yes, default is a keyword in the language.

> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/2041/files/1c9e7f5be2cf4b4aaf518cd085802c364b10b209#r64446663